### PR TITLE
Add delta-rule numerical coverage

### PR DESCRIPTION
**Pull request by drisspg**
Stacked PRs:
 * #447
 * __->__#446

--- --- ---

Add delta-rule numerical coverage

## Agent note

Add shared BF16 model-range numerical contracts for Mega GDN and KDA against independent FP64
recurrence/autograd references. Cover three deterministic seeds and every output-only gradient, and
isolate KDA's low-precision dA diagonal in a strict expected-failure test that the follow-up fix must
turn green.

## Test Plan

```bash
gpu-run --timeout 30 auto -- timeout --signal=TERM --kill-after=5s 600s pytest -q test/linear/test_delta_rule_numerics.py -p no:cacheprovider -W ignore
```

### CODE REVIEW

File: test/linear/test_delta_rule_numerics.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+import math
+
+import pytest
+import torch
+import torch.nn.functional as F
+
+pytest.importorskip(
+    "cutlass.experimental",
+    reason="the Mega numerical tests require nvidia-cutlass-dsl>=4.7",
+)
```
[the remark this answers is not in the document]
Comment (drisspg) [reply test/linear/test_delta_rule_numerics.py]: Fixed in 44c0b72: the numerical coverage now lives under test/kda/mega, so run_mega_pytest collects it in the CuTeDSL 4.7 environment. The ordinary suite treats it consistently with the existing Mega tests.

File: test/linear/test_delta_rule_numerics.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+        family, inputs, fused=False, seed=seed + 1
+    )
+
+    assert torch.isfinite(output).all()
+    assert _relative_rms(output, reference) < 1e-2
+    for name, actual, expected in zip(
+        _GRADIENT_NAMES, gradients, reference_gradients, strict=True
+    ):
+        assert torch.isfinite(actual).all(), f"{name} contains non-finite values"
+        assert _relative_rms(actual, expected) < 1e-2, name
```
[the remark this answers is not in the document]
Comment (drisspg) [reply test/linear/test_delta_rule_numerics.py]: Fixed in 44c0b72: the tests now run FP32 and FP64 recurrence oracles and use the shared assert_matches_low_precision_reference helper. I extended that helper with an RMS mode so the budget remains data-derived while avoiding poorly conditioned elementwise/max-relative comparisons.
